### ISSUE TITLE
Translate to Persian

### ISSUE THREAD

**Issue by v-zafari** [OWASP/API-Security#40]
Hello everybody,

We want to translate API-Security to Persian language.
Is there any contributions guide for this work?

**Comment by PauloASilva** [comment]
Ho @v-zafari,
Translation guidelines are still missing but it should be fairly easy.

Since you're the first person requesting a Persian translation, the issue will be assign to you.
Than it up to you to manage contributors/contributions.

To start translating the OWASP API Security Top 10 2019,

1. fork this repository,
2. create a `translation/fa` branch from `master`,
3. create a `2019/fa` directory and copy all markdown files from `2019/en` into it,
4. translate the Markdown files one by one,
5. create a `2019/fa/dist` folder and copy [the original `ODT` file][1] into it,
6. copy & paste Markdown contents into your ODT file (I am sorry but right now there's no automatic process to do this),
7. export a PDF from the ODT,
8. once everything is done and you're happy with the result, open a Pull Request to merge your `translation/fa` branch into the official repository `master` branch.

I'll be more than happy to provide you further assistance.

Cheers,
Paulo A. Silva

[1]: https://github.com/OWASP/API-Security/raw/master/2019/en/dist/owasp-api-security-top-10.odt

**Comment by v-zafari** [comment]
Thanks @PauloASilva 
The translation work has started and will end soon.

**Comment by PauloASilva** [comment]
Hi @v-zafari,
Let's keep this issue open while translation is in progress so that other guys can join you.
It would be nice to have the final document reviewed by more than one person, since we won't be able to assist you on that.

Cheers,
Paulo A. Silva

**Comment by ghookas** [comment]
May I join to this project ?
I translated it to Persian.

**Comment by PauloASilva** [comment]
@v-zafari can you please introduce @ghookas to the translation work in progress: may be together you can finish it earlier!

Cheers,
Paulo A. Silva

**Comment by ghookas** [comment]
@PauloASilva what can I do now?

**Comment by ErezYalon** [comment]
Let's give @v-zafari a few more days to respond.
@ghookas, if we get nothing by the end of the week, you can go ahead according to [these insctructions](https://github.com/OWASP/API-Security/issues/40#issuecomment-753469579) given previously by @PauloASilva.

**Comment by ghookas** [comment]
> Let's give @v-zafari a few more days to respond.
> @ghookas, if we get nothing by the end of the week, you can go ahead according to [these insctructions](https://github.com/OWASP/API-Security/issues/40#issuecomment-753469579) given previously by @PauloASilva.

I started and I will announce the result soon

**Comment by PauloASilva** [comment]
@ghookas you now own the Persian translation issue.

I've tried to reach out to @v-zafari here and by email without success.
Unfortunately I wasn't able to find public evidence of any progress on @v-zafari's fork.

Thanks for your interest and good luck.
Paulo A. Silva

**Comment by This-is-Neo** [comment]
@PauloASilva 
Hi Paulo
Since more than 6 months are passed and none of the previous efforts in translating this useful methodology to Farsi appear to be successful (or at least we were not able to find any public evidence about that), Me and my team at rnpg.ir are willing to contribute in translating it to Persian and we will prepare the final version in 3 weeks. Do we have your "go" on this?

**Comment by PauloASilva** [comment]
Hi @This-is-Neo,
We are open to contributions and I will love to review and approve such a PR.

@ghookas do you want to share the progress you've done so far?

Cheers,
Paulo A. Silva

**Comment by ghookas** [comment]
Hi @PauloASilva 
yes, I am ready to share it.

**Comment by PauloASilva** [comment]
Cool!
@This-is-Neo do you want to take this from here with @ghookas ?

Cheers,
Paulo A. Silva

**Comment by This-is-Neo** [comment]
@PauloASilva 
We are ready to collaborate by assessment and audit of the current state of the work done by @ghookas.

**Comment by PauloASilva** [comment]
@This-is-Neo please go ahead.
Unfortunately, I can't help you that much with the translation, but I'll keep watching this issue in order to provide you any further help/guidance you may need.

Cheers,
Paulo A. Silva

**Comment by ghookas** [comment]
@This-is-Neo you can pull this my branch 

https://github.com/ghookas/API-Security/tree/translation/fa/2019/fa

**Comment by ghookas** [comment]
@PauloASilva 
i can help them

**Comment by ghookas** [comment]
@This-is-Neo 
Thanks for your reply.
I don't have enough time to complete it.
I hope you and your team will do it in the perfect way.

Best Regards
Razmik

**Comment by This-is-Neo** [comment]
@ghookas Thanks for your contribution to the project, I wish you luck.

@PauloASilva Dear Paulo, considering the current state of the project, do you mind assigning the issue to me?

Regards
Reza

**Comment by PauloASilva** [comment]
Hi Folks,
I want to thank you both for the interest and energy put on this project.

@ghookas thank you very much for starting it: sometimes that is the most difficult decision.
@This-is-Neo the issue was assigned to you: this is your turn.

Best,
Paulo A. Silva

**Comment by This-is-Neo** [comment]
@PauloASilva @ErezYalon Hi
I'm glad to announce that Persian Translation is done.

**Comment by PauloASilva** [comment]
@This-is-Neo great news: congrats!

@ghookas would you like to help us reviewing/proof-reading the translation?

**Comment by ghookas** [comment]
@PauloASilva 
Yes, sure thanks for asking.

**Comment by This-is-Neo** [comment]
@ghookas We would be glad to have your opinion on the work, as soon as possible. Thanks.

**Comment by This-is-Neo** [comment]
@PauloASilva Hi
MD files are ready and suggested changes were applied.
I would be glad if you approve the pull request. thanks.
Regards
Reza

**Comment by ghookas** [comment]
> @ghookas We would be glad to have your opinion on the work, as soon as possible. Thanks.

Yes of course .

**Comment by This-is-Neo** [comment]
@PauloASilva 
@ErezYalon 
Dear Paulo and Erez! I would be glad if you could proceed with approving the pull request or inform us about possible changes that need to be applied on the work.
Thanks :)

**Comment by This-is-Neo** [comment]
@ghookas Please inform us about your opinion on the translation in 3 days, it's been more than two months that we provided the pdf of Farsi version of OWASP Top10 API and we need to conclude the work.
Thanks

**Comment by PauloASilva** [comment]
Hi,
I'll do my best to review latest changes as soon as possible and merge it.

Cheers,

**Comment by ghookas** [comment]
> @ghookas Please inform us about your opinion on the translation in 3 days, it's been more than two months that we provided the pdf of Farsi version of OWASP Top10 API and we need to conclude the work. Thanks

Hello Reza
I am going to review it and i will inform you soon.
Thanks

**Comment by ghookas** [comment]
@PauloASilva 
@This-is-Neo 

Hello All,
Thank you for giving me the chance to contribute to this project.
I congratulate you for completing this project and I must report that the translation of this document is very precise and technical and has a clear text.
Definitely, this document can be practical and useful for security and software development teams.

Thank You
Razmik

**Comment by PauloASilva** [comment]
Thank you all for giving back!

Cheers,
Paulo A. Silva

**Comment by This-is-Neo** [comment]
Dear @ghookas Thanks for your time and comments. Wish you best!
